explorer/src/commitUtils.js

```diff
@@ -0,0 +1,93 @@
+// @flow
+
+type CommitData = {
+  // TODO improve variable names
+  fileToCommits: {[filename: string]: string[]};
+  commits: {[commithash: string]: Commit};
+  authors: string[];
+}
+
+type Commit = {
+  author: string;
+  stats: {[filename: string]: FileStats};
+}
+
+type FileStats = {
+  lines: number;
+  added: number;
+  deleted: number;
+}
+
+export function commitWeight(commit: Commit, filepath: string): number {
+  // hack - GitPython encodes renames in the filepath. ignore for now.
+  if (filepath.indexOf('=>') !== -1) {
+    return 0;
+  }
+  return Math.sqrt(commit.stats[filepath].lines);
+}
+
+function allSelectedFiles(filepath: string, data: CommitData): string[] {
+  const fnames = Object.keys(data.fileToCommits);
+  return fnames.filter(x => x.startsWith(filepath));
+}
+
+function *userWeights(files: string[], data: CommitData, weightFn: WeightFn): Iterable<[string, number]> {
```

Comment:
This also merits a docstring, in my opinion; it's not clear from the interface there may be multiple tuples for a given user. (I.e., the result is not just an associative array.)